src/java.base/share/classes/jdk/internal/classfile/impl/verifier/ParserVerifier.java

```diff
@@ -0,0 +1,371 @@
+                case Utf8Entry ue -> ue::stringValue;
+                case ConstantDynamicEntry cde -> cde::asSymbol;
+                case InvokeDynamicEntry ide -> ide::asSymbol;
+                case ClassEntry ce -> ce::asSymbol;
+                case StringEntry se -> se::stringValue;
+                case MethodHandleEntry mhe -> mhe::asSymbol;
+                case MethodTypeEntry mte -> mte::asSymbol;
+                case FieldRefEntry fre -> {
+                    check.accept(fre.owner()::asSymbol);
+                    check.accept(fre::typeSymbol);
+                    yield () -> verifyFieldName(fre.name().stringValue());
+                }
+                case InterfaceMethodRefEntry imre -> {
+                    check.accept(imre.owner()::asSymbol);
+                    check.accept(imre::typeSymbol);
+                    yield () -> verifyMethodName(imre.name().stringValue());
+                }
+                case MethodRefEntry mre -> {
+                    check.accept(mre.owner()::asSymbol);
+                    check.accept(mre::typeSymbol);
+                    yield () -> verifyMethodName(mre.name().stringValue());
+                }
+                case ModuleEntry me -> me::asSymbol;
+                case NameAndTypeEntry nate -> {
+                    check.accept(nate.name()::stringValue);
+                    yield () -> nate.type().stringValue();
+                }
+                case PackageEntry pe -> pe::asSymbol;
+            });
+        }
+    }
+
+    private void verifyFieldName(String name) {
+        if (name.length() == 0 || name.chars().anyMatch(ch -> switch(ch) {
+                    case '.', ';', '[', '/' -> true;
+                    default -> false;
+                })) {
+              throw new VerifyError("Illegal field name %s in %s".formatted(name, toString(classModel)));
+        }
+    }
+
+    private void verifyMethodName(String name) {
+        if (!name.equals(INIT_NAME)
+            && !name.equals(CLASS_INIT_NAME)
+            && (name.length() == 0 || name.chars().anyMatch(ch -> switch(ch) {
+                    case '.', ';', '[', '/', '<', '>' -> true;
+                    default -> false;
+                }))) {
+              throw new VerifyError("Illegal method name %s in %s".formatted(name, toString(classModel)));
+        }
+    }
+
+    private void verifyInterfaces(List<VerifyError> errors) {
+        var intfs = new HashSet<ClassEntry>();
+        for (var intf : classModel.interfaces()) {
+            if (!intfs.add(intf)) {
+                errors.add(new VerifyError("Duplicate interface %s in %s".formatted(intf.asSymbol().displayName(), toString(classModel))));
+            }
+        }
+    }
+
+    private void verifyFields(List<VerifyError> errors) {
+        var fields = new HashSet<String>();
+        for (var f : classModel.fields()) try {
+            if (!fields.add(f.fieldName().stringValue() + f.fieldType().stringValue())) {
```

Comment:
We should declare a local record, concat is not safe if we have fields like:
```java
Loop foo;
oop fooL;
```
both will producce `fooLLoop;`

Reply:
Goo point, will fix it, thanks.